source/extensions/tracers/xray/tracer.cc

```diff
@@ -89,31 +89,34 @@ void Span::finishSpan() {
     s.mutable_http()->mutable_response()->insert(KeyValue{item.first, item.second});
   }
 
-  // TODO(marcomagdy): test how expensive this validation is. Might be worth turning off in
-  // optimized builds..
-  MessageUtil::validate(s, ProtobufMessage::getStrictValidationVisitor());
-  Protobuf::util::JsonPrintOptions json_options;
-  json_options.preserve_proto_field_names = true;
-  std::string json;
-  const auto status = Protobuf::util::MessageToJsonString(s, &json, json_options);
-  ASSERT(status.ok());
+  // TODO(marcomagdy): Test how expensive this validation is. Might be worth turning off in
+  // optimized builds.
+  try {
+    MessageUtil::validate(s, ProtobufMessage::getStrictValidationVisitor());
+  } catch (Envoy::ProtoValidationException& ex) {
+    auto& logger = Logger::Registry::getLog(Logger::Id::tracing);
+    ENVOY_LOG_TO_LOGGER(logger, trace, "X-Ray Span Validation Error: {}", ex.what());
```

Comment:
is this really trace? Shouldnt it be error?

Reply:
I've received numerous feedback from the maintainers saying "this can spam the log..". Even for configuration errors :/
This can be triggered by just passing a bad header. So it's actually high risk. I anticipate that they will have me either change it to a stat or remove it completely.

Reply:
ok :)